diff --git a/react-native-gradle-plugin/build.gradle.kts b/react-native-gradle-plugin/build.gradle.kts
index 85dadeba0c42cf0b09b964eb00b3e2c12b5ed6e9..4c8a94a8ba0e95b9cee022ca47c8629cd931432d 100644
--- a/react-native-gradle-plugin/build.gradle.kts
+++ b/react-native-gradle-plugin/build.gradle.kts
@@ -58,15 +58,15 @@ dependencies {
 // We intentionally don't build for Java 17 as users will see a cryptic bytecode version
 // error first. Instead we produce a Java 11-compatible Gradle Plugin, so that AGP can print their
 // nice message showing that JDK 11 (or 17) is required first
-java { targetCompatibility = JavaVersion.VERSION_11 }
+java { targetCompatibility = JavaVersion.VERSION_21 }

-kotlin { jvmToolchain(17) }
+kotlin { jvmToolchain(21) }

 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
     apiVersion.set(KotlinVersion.KOTLIN_1_8)
     // See comment above on JDK 11 support
-    jvmTarget.set(JvmTarget.JVM_11)
+    jvmTarget.set(JvmTarget.JVM_21)
     allWarningsAsErrors.set(
         project.properties["enableWarningsAsErrors"]?.toString()?.toBoolean() ?: false
     )
diff --git a/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt b/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt
index 151b53eaf6f28809b213cb3d76eddb084174fe6e..3f13b79b76e4dc49ac82ac47c22924514028dec7 100644
--- a/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt
+++ b/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/utils/JdkConfiguratorUtils.kt
@@ -38,8 +38,8 @@ internal object JdkConfiguratorUtils {
             project.extensions
                 .getByType(ApplicationAndroidComponentsExtension::class.java)
                 .finalizeDsl { ext ->
-                  ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_17
-                  ext.compileOptions.targetCompatibility = JavaVersion.VERSION_17
+                  ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_21
+                  ext.compileOptions.targetCompatibility = JavaVersion.VERSION_21
                 }
           }
       val libraryAction =
@@ -47,17 +47,17 @@ internal object JdkConfiguratorUtils {
             project.extensions
                 .getByType(LibraryAndroidComponentsExtension::class.java)
                 .finalizeDsl { ext ->
-                  ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_17
-                  ext.compileOptions.targetCompatibility = JavaVersion.VERSION_17
+                  ext.compileOptions.sourceCompatibility = JavaVersion.VERSION_21
+                  ext.compileOptions.targetCompatibility = JavaVersion.VERSION_21
                 }
           }
       project.pluginManager.withPlugin("com.android.application", applicationAction)
       project.pluginManager.withPlugin("com.android.library", libraryAction)
       project.pluginManager.withPlugin("org.jetbrains.kotlin.android") {
-        project.kotlinExtension.jvmToolchain(17)
+        project.kotlinExtension.jvmToolchain(21)
       }
       project.pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
-        project.kotlinExtension.jvmToolchain(17)
+        project.kotlinExtension.jvmToolchain(21)
       }
     }
   }
diff --git a/settings-plugin/build.gradle.kts b/settings-plugin/build.gradle.kts
index 2fbeafd502c98ba81856d3a1579f3b0d8bc60b2d..1709f613951b5e69a0fbcc48cb99d1e4eafbf9b4 100644
--- a/settings-plugin/build.gradle.kts
+++ b/settings-plugin/build.gradle.kts
@@ -48,15 +48,15 @@ dependencies {
 // We intentionally don't build for Java 17 as users will see a cryptic bytecode version
 // error first. Instead we produce a Java 11-compatible Gradle Plugin, so that AGP can print their
 // nice message showing that JDK 11 (or 17) is required first
-java { targetCompatibility = JavaVersion.VERSION_11 }
+java { targetCompatibility = JavaVersion.VERSION_21 }

-kotlin { jvmToolchain(17) }
+kotlin { jvmToolchain(21) }

 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
     apiVersion.set(KotlinVersion.KOTLIN_1_8)
     // See comment above on JDK 11 support
-    jvmTarget.set(JvmTarget.JVM_11)
+    jvmTarget.set(JvmTarget.JVM_21)
     allWarningsAsErrors.set(
         project.properties["enableWarningsAsErrors"]?.toString()?.toBoolean() ?: false
     )
diff --git a/settings.gradle.kts b/settings.gradle.kts
index b3c46a3bf5eb21c0bb1c9435b5582aecbc57d57a..2c0935df244682cc39346d8a77f3c6acec698701 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -13,7 +13,6 @@ pluginManagement {
   }
 }

-plugins { id("org.gradle.toolchains.foojay-resolver-convention").version("0.5.0") }

 include(
     ":react-native-gradle-plugin",
diff --git a/shared/build.gradle.kts b/shared/build.gradle.kts
index 315984875ad8ea2de476d4ab895932d2e9d47b0b..6674c4d04c0daad1beea70abc427bfad605531bf 100644
--- a/shared/build.gradle.kts
+++ b/shared/build.gradle.kts
@@ -27,15 +27,15 @@ dependencies {
   testImplementation(project(":shared-testutil"))
 }

-java { targetCompatibility = JavaVersion.VERSION_11 }
+java { targetCompatibility = JavaVersion.VERSION_21 }

-kotlin { jvmToolchain(17) }
+kotlin { jvmToolchain(21) }

 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
     apiVersion.set(KotlinVersion.KOTLIN_1_8)
     // See comment above on JDK 11 support
-    jvmTarget.set(JvmTarget.JVM_11)
+    jvmTarget.set(JvmTarget.JVM_21)
     allWarningsAsErrors.set(
         project.properties["enableWarningsAsErrors"]?.toString()?.toBoolean() ?: false
     )
diff --git a/shared-testutil/build.gradle.kts b/shared-testutil/build.gradle.kts
index 26173fb55578417118679d6243de84146b6c8a9f..42b6f6e6072e7a877b18ef2b4cc7d0e4e7a7df01 100644
--- a/shared-testutil/build.gradle.kts
+++ b/shared-testutil/build.gradle.kts
@@ -21,15 +21,15 @@ group = "com.facebook.react"

 dependencies { implementation(libs.junit) }

-java { targetCompatibility = JavaVersion.VERSION_11 }
+java { targetCompatibility = JavaVersion.VERSION_21 }

-kotlin { jvmToolchain(17) }
+kotlin { jvmToolchain(21) }

 tasks.withType<KotlinCompile>().configureEach {
   compilerOptions {
     apiVersion.set(KotlinVersion.KOTLIN_1_8)
     // See comment above on JDK 11 support
-    jvmTarget.set(JvmTarget.JVM_11)
+    jvmTarget.set(JvmTarget.JVM_21)
     allWarningsAsErrors.set(
         project.properties["enableWarningsAsErrors"]?.toString()?.toBoolean() ?: false
     )
